test/junit/scala/collection/immutable/StreamTest.scala

```diff
@@ -80,4 +80,30 @@ class StreamTest {
     assertTrue( wf.map(identity).length == 5 )       // success instead of NPE
   }
 
+  /** Test helper to verify that the given Stream operation is properly lazy in the tail */
+  def assertStreamOpLazyInTail(op: (=> Stream[Int]) => Stream[Int], expectedEvaluated: List[Int]): Unit = {
+    // Executes given effect for every strict evaluation of a Stream element
+    def naturals(f: Int => Unit): Stream[Int] = {
```

Comment:
I think this would be easier to follow if specialized for the only call-site `naturals(evaluated ++= List(_))` and renamed to `trackEffectsOnNaturals`.